core/src/main/java/org/elasticsearch/index/get/GetField.java

```diff
@@ -115,12 +115,16 @@ public static GetField fromXContent(XContentParser parser) throws IOException {
         ensureExpectedToken(XContentParser.Token.FIELD_NAME, parser.currentToken(), parser::getTokenLocation);
         String fieldName = parser.currentName();
         XContentParser.Token token = parser.nextToken();
-        ensureExpectedToken(XContentParser.Token.START_ARRAY, token, parser::getTokenLocation);
-        List<Object> values = new ArrayList<>();
-        while((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
-            values.add(parseStoredFieldsValue(parser));
+        if (token == XContentParser.Token.START_ARRAY) {
+            List<Object> values = new ArrayList<>();
+            while((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
+                values.add(parseStoredFieldsValue(parser));
+            }
+            return new GetField(fieldName, values);
+        } else {
+            parser.skipChildren();
```

Comment:
I agree, this is too lenient and also breaks the current GET behavior: the high level client ignores the 404 response code and tries to parse the response body again. If the parsing is too lenient then a potential "error" is just skipped and that's wrong. I'm looking at this.